drake/systems/frames/test/CoordinateFrameCoordinatesTest.m

```diff
@@ -0,0 +1,64 @@
+classdef CoordinateFrameCoordinatesTest < matlab.unittest.TestCase
+  %NOTEST
+  
+  % Tests that functions that touch the coordinates property of
+  % CoordinateFrame work.
+  
+  properties
+    SpecifiedFrame
+    GeneratedFrame
+  end
+  
+  methods(TestMethodSetup)
+    function createFrames (tc)
+      % SpecifiedFrame has coordinate names set by the user directly
+      tc.SpecifiedFrame = CoordinateFrame('test1', 4, 'y', {'a', 'b', 'c', 'd'});
+      
+      % GeneratedFrame has coordinate names that are
+      % automatically generated from the prefix provided.
+      tc.GeneratedFrame = CoordinateFrame('test2', 4, 'z');
+    end
+  end
+  
+  methods(Test)
+    
+    function testGetCoordinateName (tc)
```

Comment:
Re. Separate files: If you want them to be independent tests, then yes, each function would be in separate M-files

Re. Static method tests: I don't know how good of an example it is, but I used this approach in `systems/plants/collision/CollisionFilterGroup.m`. It's definitely less common in the codebase, but it is supported by `unitTest`. I like it because it keeps the tests closer to the tested code.